### ISSUE TITLE
Removing a folder that contains a file is not removed from the FileSystem Dock

### ISSUE THREAD

**Issue by Hilderin** [godotengine/godot#94431]
### Tested versions

- Reproducible in 4.3.beta.mono (97b8ad1af)
- Reproducible in 4.3 beta 3
- Not reproducible in 4.3 beta 2
- Not reproducible in 4.3 beta 1
- Not reproducible in 4.2.1

### System information

Godot v4.3.beta3 - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3060 (NVIDIA; 31.0.15.3742) - 11th Gen Intel(R) Core(TM) i5-11600KF @ 3.90GHz (12 Threads)

### Issue description

When deleting a folder using the FileSystem Dock, if the folder contains a file, everything is deleted on the disk, but in the FileSystem in the editor, onyl the file disappear. I need to restart the editor for the folder to disappear.

### Steps to reproduce

- Create a folder
- Create a scene in it
- Delete the folder in Godot

### Minimal reproduction project (MRP)

Any project.

**Comment by sci-comp** [comment]
This also happens with files for me. It's not 100% reproducible, but more often than not, I do end up having to reload the entire project as a way to force a rescan of all files (which seems to take significantly longer in 4.3 beta3, but I guess that's a separate issue).

![image](https://github.com/user-attachments/assets/0c799e83-3bd3-41b9-826d-5236860c5d1a)

**Comment by Hilderin** [comment]
@sci-comp 
Unfortunately, I did not encounter the same problem with files on my side. Did you found a pattern to reproduce the problem or maybe some actions where it happens more often like renaming, removing files; in or out of the editor?

**Comment by sci-comp** [comment]
@Hilderin 

I tried to reproduce the file error I mentioned above in a new project, but I'm unable. I added a large number of cubes from Blender thinking that a lot of clutter might help, but it does not.

https://github.com/user-attachments/assets/e6c22059-d045-46a7-95c2-72951bf58862

I will make a new issue regarding experiencing longer file system scan times if I'm able to provide more details. I think an MRP for this possible issue will need to be very large. With 200 cubes in this current example, file system scans are instant in my tests.
